execution/executor/src/chunk_executor.rs

```diff
@@ -207,12 +220,22 @@ impl<V: VMExecutor> ChunkExecutorInner<V> {
             num_txns,
         )?;
 
+        // TODO(skedia) In the chunk executor path, we ideally don't need to verify the signature
+        // as only transactions with verified signatures are committed to the storage.
+        let sig_verified_txns = SIG_VERIFY_POOL.install(|| {
```

Comment:
@msmouse, @JoshLind -  Is there any case for keeping signature verification in the state sync path? If not, I will remove this paranoid check..

Reply:
I don't think it needs to be checked -- the txns are proven to be part of the chain represented by the ledger info in the proof. The execution mode, without the signature checking, still verifies the VM executes the txns the same with the validator set, and still saves bandwidth transfering the transaction outputs directly -- just one less thing checked.

Reply:
To clarify, these are the signatures on the individual transactions themselves, right? e.g., verifying the transaction is correctly signed by the sender, etc. If so, this sounds okay to me. But, can we clarify where these checks are being done? i.e., I imagine we're still doing them in mempool, and I imagine that the VM needs to check them, too, right? Otherwise, what if we have a malicious block proposal (e.g., where some transaction signatures are invalid) where would this get caught?

Reply:
> But, can we clarify where these checks are being done? i.e., I imagine we're still doing them in mempool, and I imagine that the VM needs to check them, too, right? Otherwise, what if we have a malicious block proposal (e.g., where some transaction signatures are invalid) where would this get caught?

Yes, currently its being done twice. Once in mempool and once just before the proposal execution (although in a separate pipeline phase). 

> Likewise, what would happen in the case that someone has managed to temporarily compromise enough validators to execute incorrectly signed transactions. Would fullnodes be able to detect this if we remove these checks?

If more than f validators are compromised, then the BFT guarantees doesn't hold anymore.. not sure if FNs detecting invalid signature at that point is worth it..

Reply:
Gotcha, thanks! 😄

> If more than f validators are compromised, then the BFT guarantees doesn't hold anymore.. not sure if FNs detecting invalid signature at that point is worth it..

FWIW, I think this point is debatable. One of the big arguments around the need for "verifying read-replicas" (i.e., fullnodes) is that it improves decentralization. Sure, BFT might be compromised in the validator set, but fullnode operators will always be able to detect and call out the divergence. If this no longer becomes the case, how could one detect divergence? I'm not saying don't do it -- it's just worth thinking about the trade-off 😄

Reply:
As discussed offline, let's keep the signature verification for now. We can follow up on this.